tasks.py

```diff
@@ -0,0 +1,147 @@
+def checks(context):
+    """Run all QA checks"""
+    combined_return_code = flake8(context)
+    combined_return_code += pytest(context)
+    if combined_return_code == 0:
+        print()
+        print(r"+----------+")
+        print(r"| All good |")
+        print(r"+----------+")
+
+
+@task
+def tox(context):
+    """Run tox for the python interpreters available on your system"""
+    environments = tox_config["tox"]["envlist"].split(", ")
+
+    # Check which pythons are available on the system
+    if platform.system() == "Windows":
+        environments_to_run = filter_tox_environments_windows(environments)
+    else:
+        environments_to_run = filter_tox_environments_linux(environments)
+
+    context.run("tox -p auto -e " + ",".join(environments_to_run))
+
+
+def filter_tox_environments_linux(environments):
+    """Filter tox environments to only those available on this Linux system"""
+    environments_to_run = []
+    for environment in environments:
+        if environment.startswith("py"):
+            if environment.startswith("pypy"):
+                command = environment
+            else:
+                # The environments look like: py36
+                command = "python{}.{}".format(*environment[2: 4])
+
+            # Check that the executable exists
+            try:
+                check_call([command, "--version"])
+            except (CalledProcessError, FileNotFoundError):
+                continue
+
+            # Certain version of Ubuntu may have a old "reduced"
+            # Python version, with an "m" suffix. It is unsifficient
+            # for tox so check that it isn't there.
+            try:
+                check_call([command + "m", "--version"])
+                continue
+            except (CalledProcessError, FileNotFoundError):
+                pass
+
+            environments_to_run.append(environment)
+        else:
+            environments_to_run.append(environment)
+    return environments_to_run
+
+
+def filter_tox_environments_windows(environments):
+    """Filter tox environments to only those available on this Linux system"""
+    python_versions = []
+    for path in sys.path:
+        ppath = Path(path) / "python.exe"
+        if ppath.is_file():
+            version_string = check_output([str(ppath), "--version"]).decode("ascii").strip()
+            version_numbers = version_string.replace("Python ", "").split(".")
+            version = "".join(version_numbers[:2])
+            python_versions.append("py" + version)
+
+    environments_to_run = []
+    for environment in environments:
+        if environment.startswith("py"):
+            if environment in python_versions:
+                environments_to_run.append(environment)
+        else:
+            environments_to_run.append(environment)
+
+    return environments_to_run
+
+
+# ### Maintenance tasks
+
+
+@task
+def clean(c, dryrun=False):
+    """Clean the repository of temporary files and caches"""
```

Comment:
Fixed. ``c`` is a badly named version of the same ``context`` as before and this one has a dryrun to make it possible to see what it will delete without actually deleting it.